### PR TITLE
Consolidate RedactionGrouper and HiddenEventGrouper into MELS

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/20958

Before
image

After
image

---
Here's what your changelog entry will look like:

## ✨ Features
 * Consolidate RedactionGrouper and HiddenEventGrouper into MELS ([\#7739](https://github.com/matrix-org/matrix-react-sdk/pull/7739)). Fixes vector-im/element-web#20958.

Preview: https://62039b9ada89161a3c60a8a5--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/structures/MessagePanel.tsx [hunk 1]
```diff
@@ -1184,107 +1183,26 @@ class CreationGrouper extends BaseGrouper {
     }
 }
 
-class RedactionGrouper extends BaseGrouper {
+// Wrap consecutive member events in a ListSummary, ignore if redacted.
+// Nowadays, we also include server ACLs, pinned message changes and hidden events in this one grouper,
+// eventually we should rename it. Using multiple groupers leads to a noisier timeline.
+class MemberGrouper extends BaseGrouper {
```
Comment (SimonBrandner) [inline src/components/structures/MessagePanel.tsx:1189]: While we're here could we please rename this to not be confusing?

Comment (t3chguy) [reply src/components/structures/MessagePanel.tsx:1189]: Yeah true, I didn't think I'd consolidate Redactions at the same time, will do

**Review by gsouquet — APPROVED** [review]
Otherwise looks good to me

File: src/components/views/elements/GenericEventListSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2019, 2020 The Matrix.org Foundation C.I.C.
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/elements/GenericEventListSummary.tsx:2]: This file&class just got renamed but no implementation changes

File: src/components/views/elements/GenericEventListSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                { expanded ? _t('collapse') : _t('expand') }
+            </AccessibleButton>
+            { body }
+        </li>
+    );
+};
+
+GenericEventListSummary.defaultProps = {
+    startExpanded: false,
+    layout: Layout.Group,
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/elements/GenericEventListSummary.tsx:112]: One is done by React and the other by the language/engine. This file just got renamed so not tweaking implementation here

**Pull request merged**